client/plots/matrix.controls.js

```diff
@@ -145,6 +146,10 @@ export class MatrixControls {
 				},
 				processInput: tw => {
 					if (tw) fillTermWrapper(tw, this.opts.app.vocabApi)
+					if (tw?.term?.type == TermTypes.GENE_EXPRESSION || tw?.term?.type == TermTypes.METABOLITE_INTENSITY) {
```

Comment:
in matrix grouping by any numeric term should always be discrete, then can do isNumericTerm(tw?.term) tw.q.mode=discrete

and in what case is tw missing in this function?

please add all above comments in function

Reply:
tw missing when dividedBy term deleted.
Added comment